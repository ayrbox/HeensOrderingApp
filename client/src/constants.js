export const ORDER_TYPES = {
  delivery: 'Delivery',
  collection: 'Collection',
  table: 'Table Order',
};

export const ORDER_STATUSES = {
  ordered: 'Ordered',
  paid: 'Paid', // TODO: separate out to different status
  processing: 'Processing',
  delivered: 'Delivered',
  cancelled: 'Cancelled',
  collected: 'Collected',
  served: 'Served',
};

export const DISCOUNT_PERCENTAGES = [
  0,
  5,
  10,
  15,
  20,
  25,
];


export default {
  ORDER_TYPES,
  ORDER_STATUSES,
  DISCOUNT_PERCENTAGES,
};
