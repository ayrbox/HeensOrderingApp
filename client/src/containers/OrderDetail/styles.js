const TEXT_COLOR = '#fff';

const styles = theme => ({
  title: {
    color: TEXT_COLOR,
    paddingBottom: theme.spacing.unit,
  },
  type: {
    color: TEXT_COLOR,
    fontWeight: 'bold',
  },
  spacer: {
    marginTop: theme.spacing.unit,
    marginBottom: theme.spacing.unit,
    backgroundColor: TEXT_COLOR,
  },
  item: {
    padding: `${theme.spacing.unit}px 0 ${theme.spacing.unit}px 0`,
    justifyContent: 'space-between',
  },
  itemText: {
    color: TEXT_COLOR,
    fontSize: '0.8rem',
  },
  itemTotal: {
    color: TEXT_COLOR,
    fontSize: '0.8rem',
  },
  option: {
    padding: `0 0 0 ${theme.spacing.unit * 2}px`,
    justifyContent: 'space-between',
  },
  optionText: {
    color: TEXT_COLOR,
    fontSize: '0.65rem',
  },
  optionCost: {
    color: TEXT_COLOR,
    fontSize: '0.65rem',
  },
  orderTotal: {
    color: TEXT_COLOR,
    textAlign: 'right',
  },
});

export default styles;
