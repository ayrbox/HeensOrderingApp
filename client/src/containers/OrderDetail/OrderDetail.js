import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import Typography from '@material-ui/core/Typography';
import Divider from '@material-ui/core/Divider';
import { withStyles } from '@material-ui/core/styles';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import Button from '@material-ui/core/Button';

import { ORDER_TYPES } from '../../constants';
import styles from './styles';

const subTotal = ({
  orderItems,
}) => orderItems.reduce((_, { itemTotal }) => _ + itemTotal, 0);

const OrderDetail = ({
  classes,
  newOrder,
  showSummary,
}) => {
  const {
    orderType,
    orderItems,
  } = newOrder;

  const handleShowSummary = (e) => {
    e.preventDefault();
    showSummary();
  };

  return (
    <Fragment>
      <Typography
        variant="h6"
        className={classes.title}
      >
        Order Details
      </Typography>
      <Typography
        varaint="h1"
        className={classes.type}
      >
        {ORDER_TYPES[orderType]}
      </Typography>
      <Divider className={classes.spacer} />
      <List className={classes.items}>
        {
          orderItems.map(({ name, itemTotal, menuOptions }) => (
            <Fragment>
              <ListItem
                className={classes.item}
              >
                <Typography className={classes.itemText}>
                  {name}
                </Typography>
                <Typography className={classes.itemTotal}>
                  &pound;
                  {itemTotal.toFixed(2)}
                </Typography>
              </ListItem>
              {menuOptions.map(({ description: option, additionalCost: cost }) => (
                <ListItem
                  className={classes.option}
                >
                  <Typography className={classes.optionText}>
                    {option}
                  </Typography>
                  <Typography className={classes.itemTotal}>
                    {!!cost && `£${cost.toFixed(2)}`}
                  </Typography>
                </ListItem>
              ))}
            </Fragment>
          ))
        }
        <Divider className={classes.spacer} />
        <ListItem className={classes.item}>
          <Typography className={classes.orderTotal}>
            Total:
          </Typography>
          <Typography variant="h3" className={classes.orderTotal}>
            £
            {subTotal(newOrder).toFixed(2)}
          </Typography>
        </ListItem>
      </List>
      <Button
        variant="contained"
        color="primary"
        onClick={handleShowSummary}
      >
        Process Order
      </Button>
      {/*
        requestSuccess && msg
        && <Typography variant="h3" className={classes.msg}>{msg}</Typography>
      */}
      {/*
        !requestSuccess && msg
        && <Typography variant="h3" className={classes.errormsg}>{msg}</Typography>
      */}
    </Fragment>
  );
};

OrderDetail.propTypes = {
  classes: PropTypes.shape().isRequired,
  newOrder: PropTypes.shape({
    requestInProgress: PropTypes.bool,
    requestSuccess: PropTypes.bool,
    orderType: PropTypes.string,
    deliveryAddress: PropTypes.shape(),
    tableNo: PropTypes.string,
    orderItems: PropTypes.arrayOf(PropTypes.shape()),
    subTotal: PropTypes.number,
    discount: PropTypes.number,
    orderTotal: PropTypes.number,
    note: PropTypes.string,
  }).isRequired,
  showSummary: PropTypes.func.isRequired,
};

export default withStyles(styles)(OrderDetail);
