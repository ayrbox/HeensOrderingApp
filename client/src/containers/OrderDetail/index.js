import { connect } from 'react-redux';

import OrderDetail from './OrderDetail';
import { showSummary } from '../../actions/newOrderActions';

const mapState = state => ({
  newOrder: state.newOrder,
});

export default connect(mapState, {
  showSummary,
})(OrderDetail);
