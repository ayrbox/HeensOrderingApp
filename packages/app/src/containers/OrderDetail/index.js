import { connect } from 'react-redux';

import OrderDetail from './OrderDetail';
import { showSummary } from '../../store/newOrder';

const mapState = state => ({
  newOrder: state.newOrder,
});

export default connect(mapState, {
  showSummary,
})(OrderDetail);
