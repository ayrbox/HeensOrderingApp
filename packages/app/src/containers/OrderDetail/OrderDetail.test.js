import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import { ORDER_TYPES } from '../../constants';
import OrderDetail from './OrderDetail';


const sampleOrder = {
  requestInProgress: false,
  requestSuccess: false,
  openNewOrderPane: false,
  orderType: 'table',
  orderItems: [{
    name: 'Heen\'s Mixed Hors D\'oeuvres并',
    description: 'Mixed Hors D\'oeuvres (For 2 Persons or More)',
    price: 7.9,
    menuOptions: [{
      description: 'Chilli Sauce',
      additionalCost: 1,
    }],
    itemTotal: 8.9,
  }, {
    name: 'Wor Tip Pan Fried Pork Dumplings 鍋貼',
    description: 'Wor Tip Pan Fried Pork Dumplings (4 Pieces)',
    price: 5.5,
    menuOptions: [],
    itemTotal: 5.5,
  }, {
    name: 'Satay Chicken 串雞',
    description: 'Satay Chicken with Peanut Sauce on Skewers (4 Pieces)',
    price: 6,
    menuOptions: [],
    itemTotal: 6,
  }],
  subTotal: 20.4,
  discount: 0,
  orderTotal: 20.4,
  note: '',
  openMenuModal: false,
  openSummary: false,
  status: 'ordered',
};


describe('<OrderDetail />', () => {
  it('should render correctly', () => {
    const mockShowSummary = jest.fn();
    const {
      container,
      getByTestId,
      getAllByTestId,
    } = render(
      <OrderDetail
        newOrder={sampleOrder}
        showSummary={mockShowSummary}
      />,
    );

    expect(getByTestId(/title/)).toHaveTextContent(/Order Details/);
    expect(getByTestId(/order-type/))
      .toHaveTextContent(ORDER_TYPES[sampleOrder.orderType]);

    getByTestId('item-list');
    const items = getAllByTestId('order-item');
    expect(items).toHaveLength(sampleOrder.orderItems.length);

    const options = getAllByTestId('order-item-option');
    expect(options).toHaveLength(1);
    expect(getByTestId('order-total')).toHaveTextContent('20.4');

    fireEvent.click(getByTestId('button-process'));

    expect(mockShowSummary).toHaveBeenCalled();

    expect(container).toMatchSnapshot();
  });
});
