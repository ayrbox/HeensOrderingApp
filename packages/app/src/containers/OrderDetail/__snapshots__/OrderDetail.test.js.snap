// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderDetail /> should render correctly 1`] = `
<div>
  <h6
    class="MuiTypography-root-13 MuiTypography-h6-30 OrderDetail-title-1"
    data-testid="title"
  >
    Order Details
  </h6>
  <p
    class="MuiTypography-root-13 MuiTypography-body1-22 OrderDetail-type-2"
    data-testid="order-type"
    varaint="h1"
  >
    Table Order
  </p>
  <hr
    class="MuiDivider-root-49 OrderDetail-spacer-3"
  />
  <ul
    class="MuiList-root-54 MuiList-padding-55"
    data-testid="item-list"
  >
    <li
      class="MuiListItem-root-58 MuiListItem-default-61 MuiListItem-gutters-66 OrderDetail-item-4"
      data-testid="order-item"
    >
      <p
        class="MuiTypography-root-13 MuiTypography-body1-22 OrderDetail-itemText-5"
      >
        Heen's Mixed Hors D'oeuvres并
      </p>
      <p
        class="MuiTypography-root-13 MuiTypography-body1-22 OrderDetail-itemTotal-6"
      >
        £
        8.90
      </p>
    </li>
    <li
      class="MuiListItem-root-58 MuiListItem-default-61 MuiListItem-gutters-66 OrderDetail-option-7"
      data-testid="order-item-option"
    >
      <p
        class="MuiTypography-root-13 MuiTypography-body1-22 OrderDetail-optionText-8"
      >
        Chilli Sauce
      </p>
      <p
        class="MuiTypography-root-13 MuiTypography-body1-22 OrderDetail-itemTotal-6"
      >
        £1.00
      </p>
    </li>
    <li
      class="MuiListItem-root-58 MuiListItem-default-61 MuiListItem-gutters-66 OrderDetail-item-4"
      data-testid="order-item"
    >
      <p
        class="MuiTypography-root-13 MuiTypography-body1-22 OrderDetail-itemText-5"
      >
        Wor Tip Pan Fried Pork Dumplings 鍋貼
      </p>
      <p
        class="MuiTypography-root-13 MuiTypography-body1-22 OrderDetail-itemTotal-6"
      >
        £
        5.50
      </p>
    </li>
    <li
      class="MuiListItem-root-58 MuiListItem-default-61 MuiListItem-gutters-66 OrderDetail-item-4"
      data-testid="order-item"
    >
      <p
        class="MuiTypography-root-13 MuiTypography-body1-22 OrderDetail-itemText-5"
      >
        Satay Chicken 串雞
      </p>
      <p
        class="MuiTypography-root-13 MuiTypography-body1-22 OrderDetail-itemTotal-6"
      >
        £
        6.00
      </p>
    </li>
    <hr
      class="MuiDivider-root-49 OrderDetail-spacer-3"
    />
    <li
      class="MuiListItem-root-58 MuiListItem-default-61 MuiListItem-gutters-66 OrderDetail-item-4"
    >
      <p
        class="MuiTypography-root-13 MuiTypography-body1-22 OrderDetail-orderTotal-10"
      >
        Total:
      </p>
      <h3
        class="MuiTypography-root-13 MuiTypography-h3-27 OrderDetail-orderTotal-10"
        data-testid="order-total"
      >
        £
        20.40
      </h3>
    </li>
  </ul>
  <button
    class="MuiButtonBase-root-96 MuiButton-root-70 MuiButton-contained-81 MuiButton-containedPrimary-82 MuiButton-raised-84 MuiButton-raisedPrimary-85"
    data-testid="button-process"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label-71"
    >
      Process Order
    </span>
    <span
      class="MuiTouchRipple-root-99"
    />
  </button>
</div>
`;
